policies/api/http/logging.go

```diff
@@ -284,6 +284,20 @@ func (l loggingMiddleware) DeleteSinkFromAllDatasetsInternal(ctx context.Context
 	return l.svc.DeleteSinkFromAllDatasetsInternal(ctx, sinkID, ownerID)
 }
 
+func (l loggingMiddleware) DeleteAgentGroupFromAllDatasets(ctx context.Context, aGroupID string, token string) (err error) {
+	defer func(begin time.Time) {
+		if err != nil {
+			l.logger.Warn("method call: delete_agroup_from_all_datasets",
+				zap.Error(err),
+				zap.Duration("duration", time.Since(begin)))
+		} else {
+			l.logger.Info("method call: delete_agroup_from_all_datasets",
+				zap.Duration("duration", time.Since(begin)))
+		}
+	}(time.Now())
```

Comment:
this is an awesome way to measure the runtime, I think you could extract that to a function, you will keep the function pure, sending the function name with it, the only thing we would need to worry is the err. Let's talk about it on a separate.